gc/base/standard/ConcurrentGCSATB.hpp

```diff
@@ -79,6 +79,11 @@ class MM_ConcurrentGCSATB : public MM_ConcurrentGC
 	static MM_ConcurrentGCSATB *newInstance(MM_EnvironmentBase *env);
 	virtual void kill(MM_EnvironmentBase *env);
 
+	virtual void allocCacheRestored(MM_EnvironmentBase *env, MM_TLHAllocationSupport *tlh);
+
+	/* Refer to allocCacheRestored implementation for reasoning behind this. */
+	virtual uintptr_t reserveAllocCacheSize() { return OMR_MINIMUM_OBJECT_SIZE; }
```

Comment:
I'm on fence if those two methods should be in Collector class to start with (but simply kept within TLH code).

It is indeed semantically right to be here and it's a nice OO programming, but then, if both of methods end up having isSATBBarrierActive filter, the virtual dispatch will be completely useless.

Reply:
>should this have isSATBBarrierActive check?

Original reasoning was that if mutator is holding TLH and satb is not active and then we have Concurrent Kickoff. In this case, now we're not garentted the space when we flush if we have this condition.

Reply:
seems like not (last arg is flushCaches and it's false), but we could change it for SATB:

MM_ConcurrentGC::signalThreadsToActivateWriteBarrier(MM_EnvironmentBase *env)
...
		if (env->acquireExclusiveVMAccessForGC(this, true, false)) {

Reply:
Ah actually just realized I meant to add this `isSATBBarrierActive ` condition from the start as part of these changes... I have `GC_OMRVMInterface::flushCachesForGC(env);` in this change in `MM_ConcurrentGCSATB::setupForConcurrent`. This done right after acquiring exclusive (`if (env->acquireExclusiveVMAccessForGC(this, true, false))`

Reply:
That works, but I can remove explict flush and change condition to something like this:

```C++
if (env->acquireExclusiveVMAccessForGC(this, true, _extensions->usingSATB())
```

Not sure if this any better, since now Base class is now doing ` _extensions->usingSATB()` and overriding the the entire function doesn't seem worth it.

Reply:
Another option is to introduce a method like `bool flushCaches()` and do 

`if (env->acquireExclusiveVMAccessForGC(this, true, flushCaches())`